### ISSUE TITLE
Slice doesn't give hand back to fallback

### ISSUE THREAD

**Issue by ramnes** [qtile/qtile#614]
If you use Slice layout and cycle through it's fallback layout with `layout.{next,prev}_window()`, then after the last window the focus will come back to the sliced window.
But then `layout.{next,prev}_window()` will refuse to go back on the fallback layout and stay with the sliced window.

**Comment by tych0** [comment]
On Fri, Feb 20, 2015 at 07:32:28AM -0800, Guillaume Gelin wrote:
> If you use Slice layout and cycle through it's fallback layout with ``layout.{next,prev}_window()``, then after the last window the focus will come back to the sliced window.
> But then ``layout.{next,prev}_window()`` will refuse to go back on the fallback layout and stay with the sliced window.

Did this ever work before? If so, I probably broke it in ac45d4b0.

**Comment by ramnes** [comment]
> Did this ever work before? If so, I probably broke it in ac45d4b.

No idea, just tried Slice two minutes, seen that, then stopped using it. :eyes:

**Comment by tych0** [comment]
So after looking into this more, shouldn't the above commit fix it? ac45d4b is in 0.9.1, so if you're running at least that it should work, I think...

**Comment by ramnes** [comment]
My first post was a bit wrong though, I was trying with ``group.{next,prev}_window()`` (``group``, not ``layout``). I'm not sure what version I was running when I posted the issue, but now I'm on 0.9.1. So I just tried Slice again to check and it still acts very strangely (actually, it seems rather unusable).

I used ``layout.Slice('left', 400, wmclass='xterm', fallback=layout.Max())``, and tried to open several terminator and xterm:
- open a xterm, it goes on the left, as exepected
- open an other xterm or a terminator, it goes on second screen like if it was floating (lolwut? that looks a lot like #538)
- click back on the good screen, and windows opened in the second screen disappear (but does not appear back in the layout)
- ``group.next_window()`` makes the second window appear, but focus stays on the left xterm
- any other ``group.next_window()`` will make the Max layout change window, but the focus will always stay on the sliced window
